cli/azd/pkg/apphost/testdata/TestAspireBicepGeneration-resources.bicep.snap

```diff
@@ -63,8 +63,8 @@ resource containerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01'
   tags: tags
 }
 
-resource postgres2kv 'Microsoft.KeyVault/vaults@2023-07-01' = {
-  name: replace('postgres-2kv-${resourceToken}', '-', '')
+resource kv854251f1 'Microsoft.KeyVault/vaults@2023-07-01' = {
```

Comment:
I was sort of sad about how bad these names became, since previously the resource you'd see in the console was sort of related to some concept in the app host, and now it's not really clear what this thing is supporting.

I thought that we might be able to do better by perhaps using a shorter hash length (maybe 5 characters) and then retaining more of the original string at the start, but keyvault sets an upper limit on names to be 24 characters and the `{resourceToken}` part already takes away like 13 of them, so we are sort of running up against a wall.

Maybe we only do this hashing when the resource name (without the token) is > 11 characters (not counting the dashes which will be removed)?

Maybe this is not worthwhile since this logic will soon move into Aspire land, and we're sort of fighting against some pretty heavy constraints here.

Reply:
We still add `${resourceToken}` at the end of the unique name we produce to make it unique across the subscription :)
, so yeah, the name is super ugly.

cdk is handling this naming issue by setting a simple `take()` within the bicep template, like

```bicep
name: toLower(take(concat('sb', uniqueString(resourceGroup().id)), 24))
```

I think the will have the same issue when the resource name length is greater than 24 :D  @JoshLove-msft